### ISSUE TITLE
OBS get stuck while launching on Linux

### ISSUE THREAD

**Issue by tytan652** [obsproject/obs-studio#7095]
### Operating System Info

Other

### Other OS

Arch Linux, Flatpak both with Nvidia (not tested with another GPU)

### OBS Studio Version

28.0.0-beta2 and later

### OBS Studio Log URL

https://obsproject.com/logs/4PNy6Qu0TuGPMTaO

### Expected Behavior

OBS launch normally

### Current Behavior

Randomly, OBS will got stuck while launching and will never show a Qt window.

### Steps to Reproduce

1. Run OBS (in a terminal if possible to see that it go stuck).
2. If you are lucky the windows will never show.

### Anything else we should know?

The given log comes from #7070. But it's similar to mines when `--verbose` is not set.

Backtraces when I interrupt the program after it get stuck:
- [obs-beta2-gdboutput.log](https://github.com/obsproject/obs-studio/files/9349454/obs-beta2-gdboutput.log)
- [obs-beta2-gdboutput2.log](https://github.com/obsproject/obs-studio/files/9349455/obs-beta2-gdboutput2.log)

A trial to bisect it gave #6577 as a potential culprit but this need to be investigated further.

Beta 1 did not have this issue.

#7070 true issue can be in fact this one.

**Comment by tytan652** [comment]
It seems to get stuck in the video thread:
```
#0  __futex_abstimed_wait_common64 (private=<optimized out>, cancel=true, abstime=0x0, op=393, expected=0, futex_word=0x555556158b40) at futex-internal.c:57
57	    return INTERNAL_SYSCALL_CANCEL (futex_time64, futex_word, op, expected,
bt
#0  __futex_abstimed_wait_common64 (private=<optimized out>, cancel=true, abstime=0x0, op=393, expected=0, futex_word=0x555556158b40) at futex-internal.c:57
#1  __futex_abstimed_wait_common (futex_word=futex_word@entry=0x555556158b40, expected=expected@entry=0, clockid=clockid@entry=0, abstime=abstime@entry=0x0, private=<optimized out>, cancel=cancel@entry=true) at futex-internal.c:87
#2  0x00007ffff45a33af in __GI___futex_abstimed_wait_cancelable64 (futex_word=futex_word@entry=0x555556158b40, expected=expected@entry=0, clockid=clockid@entry=0, abstime=abstime@entry=0x0, private=<optimized out>) at futex-internal.c:139
#3  0x00007ffff45ae49f in do_futex_wait (sem=sem@entry=0x555556158b40, abstime=0x0, clockid=0) at /usr/src/debug/glibc/nptl/sem_waitcommon.c:111
#4  0x00007ffff45ae530 in __new_sem_wait_slow64 (sem=0x555556158b40, abstime=0x0, clockid=0) at /usr/src/debug/glibc/nptl/sem_waitcommon.c:183
#5  0x00007ffff45ae5a1 in __new_sem_wait (sem=<optimized out>) at sem_wait.c:42
#6  0x00007ffff58e844f in os_sem_wait (sem=<optimized out>) at /usr/src/debug/obs-studio/libobs/util/threading-posix.c:253
#7  0x00007ffff58d4af8 in video_thread (param=0x5555569c4440) at /usr/src/debug/obs-studio/libobs/media-io/video-io.c:183
#8  0x00007ffff45a678d in start_thread (arg=<optimized out>) at pthread_create.c:442
#9  0x00007ffff46278e4 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:100
```

**Comment by tytan652** [comment]
It seems stuck here:
- <https://github.com/obsproject/obs-studio/blob/master/libobs/obs.c#L3008>
- <https://github.com/obsproject/obs-studio/blob/master/libobs/util/threading-posix.c#L81>

**Comment by emaste** [comment]
As a related point, it looks like #6577 introduces a hang at exit on FreeBSD for me. (At least, building at fb146fa37084b3e88ce9199e7f0b97dedec2729d consistently exits successfully, while ef6828d31b7b5e2305c1fe810b6fed3d80fa1b46 consistently hangs).

**Comment by emaste** [comment]
My issue in #7109 is caused by f788e9f6b7d27fb30700164db6b1dde6da3cd99e

**Comment by tytan652** [comment]
I tried to reproduce it on my Intel laptop and it never got stuck. So this being a NVIDIA only issue is quiet possible.

**Comment by lolsniperftw** [comment]
Launching OBS with `QT_QPA_PLATFORM=xcb` makes OBS launch correctly so it seems with Wayland it is not showing a window. (It seems to me there is a regression because it did work fine with version 28 not long ago)
(Also on Nvidia GPU)

**Comment by tytan652** [comment]
Wrong, the issue happen on X11, Wayland was not tested because it is not supported with NVIDIA GPU by OBS.

**Comment by lolsniperftw** [comment]
I just tried on X11 and OBS is launching fine with Nvidia....

**Comment by tytan652** [comment]
Did you try to launch OBS 28 beta 2, let's say 100 times with 1 chance that OBS get stuck on start up.

When I mean "lucky", it is not a joke.

**Comment by lolsniperftw** [comment]
Ok, that is interesting since it started to work on Wayland as well and randomly failed. Also noticed it won't close when running through the terminal while the issue happens (gets stuck on shutting down)

**Comment by tytan652** [comment]
This issue is related to the startup that got stuck, every other issue should be on another github issue and NVIDIA on Wayland is not supported by OBS.

Nobody, has reported the same issue on Intel or AMD so we could guess this is NVIDIA only but it's just a guess.

**Comment by tytan652** [comment]
I finally made my `git bissect`, for each step I try to run OBS (start and exit) 100 times maximum.

```
git bisect start
git bisect bad b4f7499b33b53351e56a02fc6194f11d8af78da5     # 28 beta 2
git bisect good 4a64b2aae65dcc40e1d86ec5067532cdf9adcd35 # 28 beta 1
```
> Bisecting: 36 revisions left to test after this (roughly 5 steps)
[cb051240ffde07158c81c2963a795e9a87685410] Merge pull request #6896 from GeorgesStavracas/gbsneto/iconography

Ran OBS 100 times no issue (`git bissect good`).

> Bisecting: 18 revisions left to test after this (roughly 4 steps)
[7f3ea4e00c57db64c171c66ef30c7a3e70a34526] libobs-opengl: Check window creation for errors

Ran OBS 100 times no issue (`git bissect good`).

> Bisecting: 9 revisions left to test after this (roughly 3 steps)
[fb146fa37084b3e88ce9199e7f0b97dedec2729d] libobs: Remove display GPU markers without draws

Ran OBS 100 times no issue (`git bissect good`).

> Bisecting: 4 revisions left to test after this (roughly 2 steps)
[2c43ca1dc61e58bcdda0c62a7db63d68c88d7131] obs-ffmpeg: Add b-frame option to AMF encoder

Ran OBS 41 times and got the issue (`git bissect bad`).

> Bisecting: 2 revisions left to test after this (roughly 1 step)
warning: unable to rmdir 'plugins/obs-websocket': Directory not empty
[0a87797a219e60432616816641fe7bd97e06207b] libobs: Format changes for multiple video mixes

Had to modify `plugins/CMakeLists.txt` to ignore obs-vst.

Ran OBS 21 times and got the issue (`git bissect bad`).

> Bisecting: 0 revisions left to test after this (roughly 0 steps)
[7e39ee291cce955580b00978d2f96c391b05fb72] libobs: Add support for multiple video mixes

Ran OBS 21 times and got the issue (`git bissect bad`).

> 7e39ee291cce955580b00978d2f96c391b05fb72 is the first bad commit
